lib/msal-common/src/client/SPAClient.ts

```diff
@@ -131,47 +147,56 @@ export class SPAClient extends BaseClient {
      * also use the handleFragmentResponse() API to pass the codeResponse to this function afterwards.
      * @param codeResponse
      */
-    async acquireToken(codeResponse: CodeResponse): Promise<TokenResponse> {
-        try {
-            // If no code response is given, we cannot acquire a token.
-            if (!codeResponse || StringUtils.isEmpty(codeResponse.code)) {
-                throw ClientAuthError.createTokenRequestCannotBeMadeError();
-            }
+    async acquireToken(codeRequest: AuthorizationCodeRequest, userState: string, cachedNonce: string): Promise<TokenResponse> {
+        // If no code response is given, we cannot acquire a token.
+        if (!codeRequest || StringUtils.isEmpty(codeRequest.code)) {
+            throw ClientAuthError.createTokenRequestCannotBeMadeError();
+        }
 
-            // Get request from cache
-            const tokenRequest: TokenExchangeParameters = this.getCachedRequest(codeResponse.userRequestState);
-
-            // Initialize authority or use default, and perform discovery endpoint check.
-            const acquireTokenAuthority = (tokenRequest && tokenRequest.authority) ? AuthorityFactory.createInstance(tokenRequest.authority, this.networkClient) : this.defaultAuthority;
-            if (!acquireTokenAuthority.discoveryComplete()) {
-                try {
-                    await acquireTokenAuthority.resolveEndpointsAsync();
-                } catch (e) {
-                    throw ClientAuthError.createEndpointDiscoveryIncompleteError(e);
-                }
+        // Initialize authority or use default, and perform discovery endpoint check.
+        const acquireTokenAuthority = (codeRequest && codeRequest.authority) ? AuthorityFactory.createInstance(codeRequest.authority, this.networkClient) : this.defaultAuthority;
```

Comment:
This will change once `authority` creation is platform specific as `msal-node` does; cc @sangonzal to help @pkanher617 if needed